src/lineAuthor/lineAuthorProvider.ts

```diff
@@ -0,0 +1,100 @@
+import { Extension, Prec } from "@codemirror/state";
+import { TFile } from "obsidian";
+import { subscribeNewEditor } from "src/lineAuthor/control";
+import { eventsPerFilePathSingleton } from "src/lineAuthor/eventsPerFilepath";
+import {
+    LineAuthoring,
+    lineAuthoringId,
+    LineAuthoringId, lineAuthorState
+} from "src/lineAuthor/model";
+import { clearViewCache } from "src/lineAuthor/view/cache";
+import { lineAuthorGutter } from "src/lineAuthor/view/view";
+import ObsidianGit from "src/main";
+
+export { previewColor } from "src/lineAuthor/view/gutter/coloring";
+/**
+ * * handles changes in git head, filesystem, etc. by initiating computation
+ * * Initiates the line authoring computation via
+ * <a href="https://git-scm.com/docs/git-blame">git-blame</a>
+ * * notifies computation results and settings to subscribers (editors)
+ * * deytroys cache and editor-subscribers when plugin is deactivated
+*/
+export class LineAuthorProvider {
+    /**
+     * Saves all computed line authoring results.
+     * 
+     * See {@link LineAuthoringId}
+     */
+    private lineAuthorings: Map<LineAuthoringId, LineAuthoring> = new Map();
```

Comment:
From what I see `lineAuthorings` contains for every existed file state the blame. I think this can stack up in a longer Obsidian session. We don't need to store every blame of every ever opened file, right?

Reply:
Okay let's see if someone encounters issues.